### ISSUE TITLE
Fix S2629 FP: Allow concatenation of constants

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#8891]
### Description

S2629: Don't use string concatenation in logging message templates.

### Repro steps

```c#
            Logger.LogError(e,
                "Cannot initialize storage, will retry in {Span}. " +
                "Executed {ExecutedCount} migrations of {TotalCount}. Failed on `{Name}`. Data: {Data}",
                RetryTimeSpan,
                index,
                migrationScripts.Count,
                migrationName,
                e.Data);
```

### Expected behavior

No warnings are reported. Since I concatenate constant literals

### Actual behavior

S2629 is reported.

### Related information

* C#/VB.NET Plugins version: 9.21.0.86780

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you for reporting this. Confirmed as False Positive. I've added it to our current sprint.
